src/hb-ot-layout.cc

```diff
@@ -1703,6 +1703,59 @@ hb_ot_layout_get_size_params (hb_face_t       *face,
 
   return false;
 }
+
+struct hb_position_single_dispatch_t :
+       hb_dispatch_context_t<hb_position_single_dispatch_t, bool>
+{
+  static return_t default_return_value () { return false; }
+  bool stop_sublookup_iteration (return_t r) const { return r; }
+
+  private:
+  template <typename T, typename ...Ts> auto
+  _dispatch (const T &obj, hb_priority<1>, Ts&&... ds) HB_AUTO_RETURN
+  ( obj.position_single (std::forward<Ts> (ds)...) )
+  template <typename T, typename ...Ts> auto
+  _dispatch (const T &obj, hb_priority<0>, Ts&&... ds) HB_AUTO_RETURN
+  ( default_return_value () )
+  public:
+  template <typename T, typename ...Ts> auto
+  dispatch (const T &obj, Ts&&... ds) HB_AUTO_RETURN
+  ( _dispatch (obj, hb_prioritize, std::forward<Ts> (ds)...) )
+};
+
+hb_position_t
+hb_ot_layout_get_optical_bound (hb_font_t      *font,
+				unsigned        lookup_index,
+				hb_direction_t  direction,
+				hb_codepoint_t  glyph)
```

Comment:
Applications like LuaTeX want all the optical bound values in advance (it has a mode where line breaking takes it into account). I guess one can use `hb_ot_layout_lookup_collect_glyphs()` to get all affected glyphs and iterate on them one by one, so would it be faster if the API took an array of glyphs (or to have a different API that return optical bounds for all glyphs in the lookup?)

Reply:
Humm. That complicates things. Yeah I'm leaning towards suggesting collect_glyphs first.

Reply:
> Applications like LuaTeX want all the optical bound values in advance

We can add a version that works on an array later if that fetching one by one seems to be too slow.